### ISSUE TITLE
Why doesn't the package template include a pip-requirements file?

### ISSUE THREAD

**Issue by weaverba137** [astropy/package-template#225]
The main astropy package includes a pip-requirements file, but package-template doesn't.  Without such a file how do affiliated packages install their dependencies?  Also, how does dependency information get passed to conda?

**Comment by bsipocz** [comment]
afaik conda understands setup.py, e.g. the section ``install_requires`` lists dependencies.

**Comment by weaverba137** [comment]
OK, here's a potential bug then:  In the current setup.cfg file, `install_requires = astropy` gets converted into `['astropy']` in the setup.py file.  But it doesn't say that additional dependencies have to be *space* separated.  That is `install_requires = astropy scipy matplotlib` will work, but `install_requires = astropy,scipy,matplotlib` will not.  The specific line in setup.py is:

```python
install_requires=metadata.get('install_requires', 'astropy').strip().split(),
```

**Comment by bsipocz** [comment]
Good point. I think I would prefer ``setup.cfg`` to have ``install_requires`` in the same list format as ``setup.py`` requires.
Would you care to open a PR for this?

**Comment by bsipocz** [comment]
(The long term idea is to move all editable parts of setup.py to the config file to easy a bit of the maintenance/update burden for package maintainers).

**Comment by weaverba137** [comment]
Sure, but by "same list format" do you just mean commas, or something more explicit like:

```
install_requires = ['astropy', 'scipy', 'matplotlib']
```

**Comment by saimn** [comment]
It looks a bit weird to have a Python list inside a text config file (whose goal to have simpler syntax imo). Why not a comma-separated list ? It seems the most intuitive.

**Comment by bsipocz** [comment]
@saimn - I thought it's the least ambiguous option, as noted above, the original setup expected a space separated list. 
However none of the 100+ packages I have a local copy use this option yet, so we can still choose the expected format and I'm happy to have whichever more people like.

@eteq @adrn @astrofrog - Do you have any preferences?

**Comment by astrofrog** [comment]
Isn't there a standard for config files? (if parsed by e.g. configobj). I imagine it would be comma-separated but without square brackets?

**Comment by bsipocz** [comment]
OK, then it's pretty clear that originally we have a bug, and the list should be a coma separated list as @saimn suggested. 

http://configobj.readthedocs.io/en/latest/configobj.html#the-config-file-format

Could you @weaverba137 update the PR accordingly, please?

**Comment by weaverba137** [comment]
Ah, let's not get too hasty here.  The config.cfg file is parsed by configparser, not configobj.

**Comment by saimn** [comment]
Yep, configobj is not the same thing as configparser. configparser is in the standard library, but basically it does not try to parse values.  So, you can choose the format that seems the best:

- space-separated is actually not so bad, maybe it just needs to be better documented (with a comment in the example setup.cfg ?) + it avoids to break compatibility
- comma-separated, 
- another possibility is a multiline value, but I'm not sure that Python2's version of configparser supports this (the module was improved in Python3 and there is a backport, was maybe overkill here and having an additional dependency for the setup is never pratical) 
- or Python-like list

**Comment by weaverba137** [comment]
OK, in the interest of moving ahead here, can we shift this discussion over to the specific changes proposed in #226?  I will note here that @saimn's first suggestion above, a space-separated list, would be the absolute minimal change, and would require no changes to the existing setup.py file at all.  If more people favor that, please comment on #226.

**Comment by weaverba137** [comment]
Closed by #226.
